common/network-common/src/main/java/org/apache/spark/network/util/RocksDBProvider.java

```diff
@@ -0,0 +1,170 @@
+
+        dbOptions.setCreateIfMissing(false);
+        dbOptions.setBottommostCompressionType(CompressionType.ZSTD_COMPRESSION);
+        dbOptions.setCompressionType(CompressionType.LZ4_COMPRESSION);
+        dbOptions.setTableFormatConfig(tableFormatConfig);
+        dbOptions.setLogger(rocksDBLogger);
+
+        try {
+          tmpDb = RocksDB.open(dbOptions, dbFile.toString());
+        } catch (RocksDBException e) {
+          if (e.getStatus().getCode() == Status.Code.NotFound) {
+            logger.info("Creating state database at " + dbFile);
+            dbOptions.setCreateIfMissing(true);
+            try {
+              tmpDb = RocksDB.open(dbOptions, dbFile.toString());
+            } catch (RocksDBException dbExc) {
+              throw new IOException("Unable to create state store", dbExc);
+            }
+          } else {
+            // the RocksDB file seems to be corrupt somehow.  Let's just blow it away and create
+            // a new one, so we can keep processing new apps
+            logger.error("error opening rocksdb file {}. Creating new file, will not be able to " +
+              "recover state for existing applications", dbFile, e);
+            if (dbFile.isDirectory()) {
+              for (File f : Objects.requireNonNull(dbFile.listFiles())) {
+                if (!f.delete()) {
+                  logger.warn("error deleting {}", f.getPath());
+                }
+              }
+            }
+            if (!dbFile.delete()) {
+              logger.warn("error deleting {}", dbFile.getPath());
+            }
+            dbOptions.setCreateIfMissing(true);
+            try {
+              tmpDb = RocksDB.open(dbOptions, dbFile.toString());
+            } catch (RocksDBException dbExc) {
+              throw new IOException("Unable to create state store", dbExc);
+            }
+          }
+        }
+        try {
+          // if there is a version mismatch, we throw an exception, which means the service
+          // is unusable
+          checkVersion(tmpDb, version, mapper);
+        } catch (RocksDBException e) {
+          throw new IOException(e.getMessage(), e);
+        }
+      }
+      return tmpDb;
+    }
+
+    @VisibleForTesting
+    static RocksDB initRocksDB(File file) throws IOException {
+      BloomFilter fullFilter =
+        new BloomFilter(10.0D /* BloomFilter.DEFAULT_BITS_PER_KEY */, false);
+      BlockBasedTableConfig tableFormatConfig = new BlockBasedTableConfig()
+        .setFilterPolicy(fullFilter)
+        .setEnableIndexCompression(false)
+        .setIndexBlockRestartInterval(8)
+        .setFormatVersion(5);
+
+      Options dbOptions = new Options();
+      dbOptions.setCreateIfMissing(false);
+      dbOptions.setBottommostCompressionType(CompressionType.ZSTD_COMPRESSION);
+      dbOptions.setCompressionType(CompressionType.LZ4_COMPRESSION);
+      dbOptions.setTableFormatConfig(tableFormatConfig);
+      try {
+        return RocksDB.open(dbOptions, file.toString());
```

Comment:
Why don't we have `setCreateIfMissing` resetting as we do in ` initRockDB(File dbFile, StoreVersion version, ObjectMapper mapper)`? Is the call on this method supposed to be the first time RocksDB creation?

Reply:
Line 58 set `dbOptions.setCreateIfMissing(false)` first and want to restore an existing RocksDB instance.

If the restore catch `RocksDBException`,  set  `dbOptions.setCreateIfMissing(true)` enter the process of creating a new RocksDB instance.

Reply:
Sorry, this method just use by test

Reply:
https://github.com/apache/spark/blob/43b02d7f25b2ab99f9b0be3c45aaedb1a8bdd40c/resource-managers/yarn/src/test/scala/org/apache/spark/network/shuffle/ShuffleTestAccessor.scala#L210-L217

Reply:
move to a new Test Utils?

Reply:
@Ngone51 https://github.com/apache/spark/pull/37648 unified the use of `initDB` method. If it is merged, this method can be deleted.